Build failure real. You'll have to disable those tests and should comment on the utils that they are testing since they won't work for prod after the migration:
```
ERROR: corehq.sql_db.tests.test_shard_management:ShardManagementTest.test_text_partitioning_correct
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/sql_db/tests/test_shard_management.py", line 71, in test_text_partitioning_correct
    matches = get_count_of_unmatched_models_by_shard(self.db2, XFormInstanceSQL)
  File "/mnt/commcare-hq-ro/corehq/sql_db/shard_data_management.py", line 97, in get_count_of_unmatched_models_by_shard
    cursor.execute(query, [valid_shards])
  File "/vendor/lib/python2.7/site-packages/raven/contrib/django/client.py", line 114, in execute
    return real_execute(self, sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
ProgrammingError: function hash_string(character varying, unknown) does not exist
LINE 3:         select hash_string(form_id, 'siphash24') & 3 as shar...
```

--- a/corehq/form_processor/backends/sql/dbaccessors.py
+++ b/corehq/form_processor/backends/sql/dbaccessors.py
@@ -104,7 +104,7 @@ class ShardAccessor(object):
     hash_key = b'\x00' * 16
 
     @staticmethod
-    def hash_doc_ids_sql(doc_ids):
+    def hash_doc_ids_sql_for_testing(doc_ids):
         """Get HASH for each doc_id from PostgreSQL
 
         This is used to ensure the python version is consistent with what's
@@ -123,7 +123,7 @@ class ShardAccessor(object):
             return {row.doc_id: row.hash for row in rows}
 
     @staticmethod
-    def hash_doc_uuid_sql(doc_uuid):
+    def hash_doc_uuid_sql_for_testing(doc_uuid):
         """Get the hash for a UUID from PostgreSQL
 
         This is used to ensure the python version is consistent with what's

--- a/corehq/form_processor/tests/test_sharding.py
+++ b/corehq/form_processor/tests/test_sharding.py
@@ -148,7 +148,7 @@ class ShardAccessorTests(TestCase):
     def test_hash_doc_ids(self):
         N = 1001
         doc_ids = [str(i) for i in range(N)]
-        hashes = ShardAccessor.hash_doc_ids_sql(doc_ids)
+        hashes = ShardAccessor.hash_doc_ids_sql_for_testing(doc_ids)
         self.assertEquals(len(hashes), N)
         self.assertTrue(all(isinstance(hash_, int) for hash_ in hashes.values()))
 
@@ -175,7 +175,7 @@ class ShardAccessorTests(TestCase):
         N = 2048
         doc_ids = [str(i) for i in range(N)]
 
-        sql_hashes = ShardAccessor.hash_doc_ids_sql(doc_ids)
+        sql_hashes = ShardAccessor.hash_doc_ids_sql_for_testing(doc_ids)
 
         csiphash_hashes = ShardAccessor.hash_doc_ids_python(doc_ids)
         self.assertEquals(len(csiphash_hashes), N)
@@ -192,4 +192,4 @@ class ShardAccessorTests(TestCase):
     def test_hash_uuid(self):
         uuid = UUID('403724ef9fe141f2908363918c62c2ff')
         self.assertEqual(ShardAccessor.hash_doc_id_python(uuid), 1415444857)
-        self.assertEqual(ShardAccessor.hash_doc_uuid_sql(uuid), 1415444857)
+        self.assertEqual(ShardAccessor.hash_doc_uuid_sql_for_testing(uuid), 1415444857)

--- a/corehq/sql_accessors/migrations/0056_add_hashlib_functions.py
+++ b/corehq/sql_accessors/migrations/0056_add_hashlib_functions.py
@@ -3,7 +3,8 @@
 from __future__ import absolute_import, unicode_literals
 
 from django.db import migrations
-from corehq.sql_db.operations import noop_migration
+from django.conf import settings
+from corehq.sql_db.operations import HqRunSQL, noop_migration
 
 
 class Migration(migrations.Migration):
@@ -13,8 +14,12 @@ class Migration(migrations.Migration):
     ]
 
     operations = [
-        # this originally installed the hashlib extension
+        # this originally installed the hashlib extension in production as well
         # but commcare-cloud does that where possible already
         # and Amazon RDS doesn't allow it
-        noop_migration()
+        HqRunSQL(
+            'CREATE EXTENSION IF NOT EXISTS hashlib',
+            'DROP EXTENSION hashlib'
+        )
+        if settings.UNIT_TESTING else noop_migration()
     ]

--- a/corehq/sql_db/management/commands/print_database_shard_data.py
+++ b/corehq/sql_db/management/commands/print_database_shard_data.py
@@ -1,7 +1,7 @@
 from __future__ import absolute_import, print_function
 from __future__ import unicode_literals
 from django.core.management.base import BaseCommand
-from corehq.sql_db.shard_data_management import get_database_shard_info
+from corehq.sql_db.shard_data_management import get_database_shard_info_for_testing
 from corehq.sql_db.util import get_db_aliases_for_partitioned_query
 
 
@@ -30,7 +30,7 @@ class Command(BaseCommand):
         for database in get_db_aliases_for_partitioned_query():
             if verbose:
                 print('Checking database {}...'.format(database))
-            shard_info = get_database_shard_info(database)
+            shard_info = get_database_shard_info_for_testing(database)
             if options['csv']:
                 print(shard_info.to_csv())
             else:

--- a/corehq/sql_db/shard_data_management.py
+++ b/corehq/sql_db/shard_data_management.py
@@ -71,11 +71,11 @@ class DatabaseShardInfo(object):
         return '\n'.join(formatted_rows)
 
 
-def get_database_shard_info(database):
+def get_database_shard_info_for_testing(database):
     sharded_models = list(get_all_sharded_models())
     shard_info = DatabaseShardInfo(database)
     for model in sharded_models:
-        data = get_count_of_models_by_shard(database, model)
+        data = get_count_of_models_by_shard_for_testing(database, model)
         shard_info.add_model_data(model, data)
     return shard_info
 
@@ -92,22 +92,22 @@ def get_count_of_unmatched_models_by_shard(database, model):
     The list will be empty if no invalid data is found.
     """
     cursor = connections[database].cursor()
-    query = _get_unmatched_shard_count_query(model)
+    query = _get_unmatched_shard_count_query_for_testing(model)
     valid_shards = partition_config.get_shards_on_db(database)
     cursor.execute(query, [valid_shards])
     results = cursor.fetchall()
     return results
 
 
-def get_count_of_models_by_shard(database, model):
+def get_count_of_models_by_shard_for_testing(database, model):
     cursor = connections[database].cursor()
-    query = _get_counts_by_shard_query(model)
+    query = _get_counts_by_shard_query_for_testing(model)
     cursor.execute(query)
     results = cursor.fetchall()
     return results
 
 
-def _get_unmatched_shard_count_query(model):
+def _get_unmatched_shard_count_query_for_testing(model):
     # syntax of this query is a bit weird because of a couple django / postgres ARRAY oddities
     # https://stackoverflow.com/a/22008870/8207
     # https://stackoverflow.com/a/11730789/8207
@@ -117,11 +117,11 @@ def _get_unmatched_shard_count_query(model):
         ) as countsByShard
         where not shard_id = ANY(%s);
     """.format(
-        counts_by_shard_query=_get_counts_by_shard_query(model),
+        counts_by_shard_query=_get_counts_by_shard_query_for_testing(model),
     )
 
 
-def _get_counts_by_shard_query(model):
+def _get_counts_by_shard_query_for_testing(model):
     # have to cast to varchar because some tables have uuid types
     field_type = model._meta.get_field(model.partition_attr)
     if isinstance(field_type, UUIDField):

--- a/corehq/sql_db/tests/test_shard_management.py
+++ b/corehq/sql_db/tests/test_shard_management.py
@@ -15,7 +15,7 @@ from corehq.messaging.scheduling.scheduling_partitioned.tests.test_dbaccessors_p
 from corehq.sql_db.config import partition_config
 from corehq.sql_db.models import PartitionedModel
 from corehq.sql_db.shard_data_management import get_count_of_unmatched_models_by_shard, \
-    get_count_of_models_by_shard
+    get_count_of_models_by_shard_for_testing
 from corehq.sql_db.tests.utils import DefaultShardingTestConfigMixIn
 
 
@@ -50,7 +50,7 @@ class ShardManagementTest(DefaultShardingTestConfigMixIn, TestCase):
         self.assertEqual(AlertScheduleInstance.objects.using(self.db1).count(), 1)
         matches = get_count_of_unmatched_models_by_shard(self.db1, AlertScheduleInstance)
         self.assertEqual(0, len(matches))
-        all_data = get_count_of_models_by_shard(self.db1, AlertScheduleInstance)
+        all_data = get_count_of_models_by_shard_for_testing(self.db1, AlertScheduleInstance)
         self.assertEqual(1, len(all_data))
         self.assertEqual((0, 1), all_data[0])
 
@@ -70,7 +70,7 @@ class ShardManagementTest(DefaultShardingTestConfigMixIn, TestCase):
         self.assertEqual(XFormInstanceSQL.objects.using(self.db2).count(), 1)
         matches = get_count_of_unmatched_models_by_shard(self.db2, XFormInstanceSQL)
         self.assertEqual(0, len(matches))
-        all_data = get_count_of_models_by_shard(self.db2, XFormInstanceSQL)
+        all_data = get_count_of_models_by_shard_for_testing(self.db2, XFormInstanceSQL)
         self.assertEqual(1, len(all_data))
         self.assertEqual((2, 1), all_data[0])
 

